src/peripherals.rs

```diff
@@ -202,30 +259,225 @@ where
+            rx_grant.set_pipe(pipe);
+            rx_grant.set_rssi(rssi);
+            rx_grant.commit_all();
+        } else {
+            // Drop `tx_grant` and `rx_grant` so the upper stack can pass them again in the next
+            // `transmit` and `prepare_for_ack`
+            self.tx_grant.take();
+            self.rx_grant.take();
+        }
         ret
     }
 
-    // Disables the radio and the `radio disabled` interrupt
-    pub(crate) fn stop(&mut self) {
-        self.radio.intenclr.write(|w| w.disabled().set_bit());
-        self.radio.tasks_disable.write(|w| unsafe { w.bits(1) });
+    // --------------- PRX methods --------------- //
 
-        // Wait for the disable event to kick in, to make sure that the `task_disable` write won't
-        // trigger an interrupt
-        while self.radio.events_disabled.read().bits() == 0 {}
-        self.clear_disabled_event();
+    // Start listening for packets and setup necessary shorts and interrupts
+    pub(crate) fn start_receiving(&mut self, mut rx_buf: PayloadW<IncomingLen>, enabled_pipes: u8) {
+        // Start TX after receiving a packet as it might need an ack
+        self.radio.shorts.modify(|_, w| w.disabled_txen().enabled());
+
+        self.radio.intenset.write(|w| w.disabled().set_bit());
+        self.radio
+            .rxaddresses
+            .write(|w| unsafe { w.bits(enabled_pipes as u32) });
+
+        unsafe {
+            self.radio
+                .packetptr
+                .write(|w| w.bits(rx_buf.dma_pointer() as u32));
+            self.radio.events_address.write(|w| w.bits(0));
+            self.clear_disabled_event();
+            self.clear_ready_event();
+            self.clear_end_event();
+            //self.radio.events_payload.write(|w| w.bits(0)); do we need this ? Probably not
+
+            // "Preceding reads and writes cannot be moved past subsequent writes."
+            compiler_fence(Ordering::Release);
 
+            self.radio.tasks_rxen.write(|w| w.bits(1));
+        }
+        self.rx_grant = Some(rx_buf);
+    }
+
+    // Check the received packet.
+    //
+    // # Safety
+    //
+    // This function is non-reentrant
+    pub(crate) unsafe fn check_packet(
+        &mut self,
+        mut payload: Option<PayloadR<OutgoingLen>>,
+    ) -> RxPayloadState {
+        // If the user didn't provide a packet to send, we will fall back to this empty ack packet
+        static FALLBACK_ACK: [u8; 2] = [0, 0];
+
+        // We can't know if our last ack was received before entering this method, and because
+        // bbqueue doesn't support having multiple grants at the same time, we will need to cache
+        // the last packet, otherwise we would only be able to send user provided packets at a
+        // maximum rate of one packet per two transactions, even when all our acks are being
+        // received. This doesn't seem like the perfect solution.
+        static mut CACHE: [u8; 254] = [0u8; 254];
```

Comment:
Ok, this solution is not very nice, but I couldn't find a better one. I'm open for feedback.

Reply:
> bbqueue doesn't support having multiple grants at the same time

Okay, I've had a couple people ask me for this, so I think I'm going to implement it. I'm thinking I can offer:

* A grant that holds access to multiple frames
* You can commit N frames at a time
* Maybe impl iterator for the multi-frame item?

Reply:
I found a better solution that doesn't involving copying. It's actually quite simple, it was late, heh.